src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,138 @@
+            if (ensureCapacity(ammountOfElements + 1)) {
+                System.arraycopy(elementData, index, elementData,
+                        index + 1, ammountOfElements + 1 - index);
+                elementData[index] = value;
+                ammountOfElements++;
+            } else {
+                Object[] oldData = elementData;
+                newArrayCapasity(arrayCapasity);
+                System.arraycopy(oldData, index, elementData,
+                        index + 1, ammountOfElements + 1 - index);
+                elementData[index] = value;
+                ammountOfElements++;
+            }
+        } else {
+            throw new ArrayIndexOutOfBoundsException("ammountOfElements: "
+                    + ammountOfElements + "<" + "index: " + index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (ensureCapacity(ammountOfElements + list.size())) {
+            for (int i = 0; i < list.size(); i++) {
+                elementData[ammountOfElements++] = list.get(i);
+            }
+        } else {
+            Object[] oldData = elementData;
+            newArrayCapasity(ammountOfElements + list.size());
+            System.arraycopy(oldData, 0, elementData, 0, ammountOfElements + 1);
+            for (int i = 0; i < list.size(); i++) {
+                elementData[ammountOfElements++] = list.get(i);
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < ammountOfElements) {
+            return (T) elementData[index];
+        } else {
+            throw new ArrayIndexOutOfBoundsException("ammountOfElements: "
+                    + ammountOfElements + "<" + "index: " + index);
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < ammountOfElements || index < 0) {
+            elementData[index] = value;
+        } else {
+            throw new ArrayIndexOutOfBoundsException("ammountOfElements: "
+                    + ammountOfElements + "<" + "index: " + index);
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < ammountOfElements || index < 0) {
+            int numMoved = ammountOfElements - index - 1;
+            T result = (T) elementData[index];
+            System.arraycopy(elementData, index + 1, elementData, index, numMoved);
+            elementData[--ammountOfElements] = null;
+            return result;
+        } else {
+            throw new ArrayIndexOutOfBoundsException("ammountOfElements: "
+                    + ammountOfElements + "<" + "index: " + index);
+        }
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < ammountOfElements; i++) {
+            if ((t == null && elementData[i] == null)
+                    || (elementData[i] != null && elementData[i].equals(t))) {
+                int numMoved = ammountOfElements - i - 1;
```

Comment:
when you found your element just call your remove(int index) method, cause same code repeats